backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java

```diff
@@ -282,8 +281,9 @@ public CommitEntity getCommitEntity(
         throw new ModelDBException("Commit is associated with Label", Code.FAILED_PRECONDITION);
       }
 
+      session.beginTransaction();
```

Comment:
I believe it's fine to leave the logic above inside the transaction. It should be pretty fast. We can revisit if not.

Reply:
reverted as per your comment